### DISCUSSION TITLE
Different values using Google and Python to calculate areas and perimeters

### DISCUSSION THREAD

**Discussion by diegodab** [pyproj4/pyproj#914] [Q&A]
Hello,

Both use the WGS 84 (World Geodetic System: https://en.wikipedia.org/wiki/World_Geodetic_System#WGS84), and selecting exactly the same coordinates there is around 0.1% difference on the values.

The following code, written in Python, calculates the area and perimeter based on latitude and altitude. The Geod function used in the pyproj library of Python uses WGS84 too.

```
from pyproj import Geod
geod = Geod(ellps='WGS84')

lons = [-3.589221363867642, 2.174356149209656, -3.6883348421485063]
lats = [37.177242962056454, 41.40363110596998, 40.453039980359634]
poly_area, poly_perimeter = geod.polygon_area_perimeter(lons, lats)

print("Area:", (abs(float(poly_area))) / (1000000), "km^2")
print("Perimeter:", (abs(float(poly_perimeter))) / (1000), "km")
```

Do you know what causes the difference, if we use the same coordinates and the geoid is identical?

**Comment by snowman2** [comment]
I wonder if this is related: https://gis.stackexchange.com/q/331279

**Reply by diegodab** [reply]
Hello, I am afraid I always choose  the points with as much precision as I can. The error sometimes is about 0.06%: very small and for some applications would not be a problem, but I requiere extreme precision and I don't get why there is difference between Google and my code, if we use the same coordinates (up to 15 decimals).

**Reply by snowman2** [reply]
It is hard to know which one is different or why. I am not sure how Google calculates things behind the scenes, so I cannot answer why there is a difference. However, you can go to https://lists.osgeo.org/mailman/listinfo/proj/ and ask how PROJ calculates geodesic area and/or look into the code https://github.com/osgeo/PROJ/ to help your confidence in the response from pyproj.
